### PR TITLE
TE-264 Use checked arithmetics for encoded size

### PR DESCRIPTION
**Pull request by akoptelov**
Not sure if it is possible to add any tests for testing usize overflow though...

### CODE REVIEW

File: tezos/encoding/src/binary_writer.rs [hunk 1]
```diff
@@ -472,6 +472,16 @@ fn find_value_in_record_values<'a>(
...
 }
 
+fn encoded_size_add(size: usize, val: usize) -> Result<usize, Error> {
+    size.checked_add(val)
+        .ok_or(Error::custom("Encoded message size overflow"))
+}
+
+fn encoded_size_sub(size: usize, val: usize) -> Result<usize, Error> {
+    size.checked_sub(val)
+        .ok_or(Error::custom("Encoded message size overflow"))
```
Comment (bkontur) [inline tezos/encoding/src/binary_writer.rs:482]: @akoptelov 
could you please just change: overflow -> underflow?

Comment (akoptelov) [reply tezos/encoding/src/binary_writer.rs:482]: @bkontur From CPU point of view this is still overflow, see the std documentation: https://doc.rust-lang.org/std/primitive.usize.html#method.checked_sub

Comment (bkontur) [reply tezos/encoding/src/binary_writer.rs:482]: @akoptelov 
aha, ok, ok, I just want to have different error messages for different cases - that we could differ directly from the log in case of error

Comment (akoptelov) [reply tezos/encoding/src/binary_writer.rs:482]: To my understanding this is very unlikely that this ever happen. And if it will, it will be quite hard reproducing it with backtrace option. So probably we can use different messages for each place instead. I'll do that.

Comment (akoptelov) [reply tezos/encoding/src/binary_writer.rs]: Please take a look if there is enough information now for possible troubleshooting.

Comment (bkontur) [reply tezos/encoding/src/binary_writer.rs]: cool, looks ok :)

**Review by bkontur — APPROVED** [review]

**Comment by bkontur** [comment]
@akoptelov 
merged, close please jira
